ipn/store/kubestore/store_kube.go

```diff
@@ -144,9 +144,15 @@ func (s *Store) WriteTLSCertAndKey(domain string, cert, key []byte) (err error)
 		return fmt.Errorf("invalid domain name %q: %w", domain, err)
 	}
 	defer func() {
-		// TODO(irbekrm): a read between these two separate writes would
-		// get a mismatched cert and key.  Allow writing both cert and
-		// key to the memory store in a single, lock-protected operation.
+		// TODO(irbekrm): certs for write replicas are currently not
+		// written to memory to avoid out of sync memory state after
+		// Ingress resources have been recreated.  This means that TLS
+		// certs for write replicas are retrieved from the Secret on
+		// each HTTPS request.  This is a temporary solution till we
+		// implement a Secret watch.
+		if s.certShareMode == "rw" {
```

Comment:
nit: This approach minimises the diff/indentation but I think it would keep the control flow simpler to avoid calling the deferred function at all.

Reply:
true